tests/migrations/downgrade_test.py

```diff
@@ -132,3 +169,283 @@ def get_migrate_version(db_engine: Engine):
+                )
+            ).scalar()
+
+            # Ensure that there are 3 records.
+            #    (token_address1, exchange_address1)
+            #    (token_address1, exchange_address2)
+            #    (token_address2, exchange_address2)
+            assert all_row_count == 3
+            assert token_address_1_count == 2
+
+            # NOTE: idx_position_share_block_number
+            all_row_count = conn.execute(
+                text("SELECT COUNT(*) FROM idx_position_share_block_number")
+            ).scalar()
+            token_address_1_count = conn.execute(
+                text(
+                    "SELECT COUNT(*) FROM idx_position_share_block_number WHERE token_address = 'token_address1'"
+                )
+            ).scalar()
+
+            # Ensure that there are 3 records.
+            #    (token_address1, exchange_address1)
+            #    (token_address1, exchange_address2)
+            #    (token_address2, exchange_address2)
+            assert all_row_count == 3
+            assert token_address_1_count == 2
+
+            # NOTE: idx_position_coupon_block_number
+            all_row_count = conn.execute(
+                text("SELECT COUNT(*) FROM idx_position_coupon_block_number")
+            ).scalar()
+            token_address_1_count = conn.execute(
+                text(
+                    "SELECT COUNT(*) FROM idx_position_coupon_block_number WHERE token_address = 'token_address1'"
+                )
+            ).scalar()
+
+            # Ensure that there are 3 records.
+            #    (token_address1, exchange_address1)
+            #    (token_address1, exchange_address2)
+            #    (token_address2, exchange_address2)
+            assert all_row_count == 3
+            assert token_address_1_count == 2
+
+            # NOTE: idx_position_membership_block_number
+            all_row_count = conn.execute(
+                text("SELECT COUNT(*) FROM idx_position_membership_block_number")
+            ).scalar()
+            token_address_1_count = conn.execute(
+                text(
+                    "SELECT COUNT(*) FROM idx_position_membership_block_number WHERE token_address = 'token_address1'"
+                )
+            ).scalar()
+
+            # Ensure that there are 3 records.
+            #    (token_address1, exchange_address1)
+            #    (token_address1, exchange_address2)
+            #    (token_address2, exchange_address2)
+            assert all_row_count == 3
+            assert token_address_1_count == 2
+
+        for assert_table_name in table_name_list:
+            table_bf = self.create_sorted_table(tables_bf[assert_table_name])
+            table_af = self.create_sorted_table(tables_af[assert_table_name])
+            assert (
+                CreateTable(table_af).compile(engine).string
+                == CreateTable(table_bf).compile(engine).string
+            )
```

Comment:
- Added code to verify that the following two types of databases are identical:
  - 1.database that has been upgraded to version **23.3 after initialization**
  - 2.database that has been upgraded to version 23.6 and then **downgraded to version 23.3**